include/openPMD/RecordComponent.hpp

```diff
@@ -124,24 +140,16 @@ RecordComponent::loadChunk(Offset const& o, Extent const& e, std::shared_ptr< T
                                      + " - DS: " + std::to_string(dse[i])
                                      + " - Chunk: " + std::to_string(o[i] + e[i])
                                      + ")");
-    if( Allocation::API == alloc && data )
-        throw std::runtime_error("Preallocated pointer passed with signaled API-allocation during chunk loading.");
-    else if( Allocation::USER == alloc && !data )
-        throw std::runtime_error("Unallocated pointer passed with signaled user-allocation during chunk loading.");
+    if( !data )
+        throw std::runtime_error("Unallocated pointer passed during chunk loading.");
 
-    uint64_t numPoints = 1u;
-    for( auto const& dimensionSize : e )
-        numPoints *= dimensionSize;
-
-    if( (Allocation::AUTO == alloc && !data) || Allocation::API == alloc )
-    {
-        auto newData = std::shared_ptr<T>( new T[numPoints], []( T *p ){ delete [] p; } );
-        data.swap(newData);
-    }
     T* raw_ptr = data.get();
 
     if( *m_isConstant )
     {
+        uint64_t numPoints = 1u;
+        for( auto const& dimensionSize : e )
+            numPoints *= dimensionSize;
         /*
```

Comment:
This block of code can also be removed. The constant value is cached within the ```RecordComponent``` itself now.